openedx/core/lib/xblock_builtin/xblock_discussion/tests.py

```diff
@@ -0,0 +1,136 @@
+        self.id_gen_mock = mock.Mock()
+
+    def _construct_xblock_mock(self, cls, keys):  # pylint: disable=unused-argument
+        """
+        Builds target xblock instance (DiscussionXBlock)
+
+        Signature-compatible with runtime.construct_xblock_from_class - can be used as a mock side-effect
+        """
+        return DiscussionXBlock(self.runtime_mock, scope_ids=keys, field_data=DictFieldData({}))
+
+    @mock.patch(DISCUSSION_XBLOCK_LOCATION + ".load_definition_xml")
+    @ddt.unpack
+    @ddt.data(
+        ("ID1", "Test category", "Test target"),
+        ("ID2", "Some other category", "Some other target")
+    )
+    def test_xblock_export_format(self, discussion_id, category, target, patched_load_definition_xml):
+        """
+        Test that xblock export XML format can be parsed preserving field values
+        """
+        xblock_xml = """
+        <discussion
+            url_name="82bb87a2d22240b1adac2dfcc1e7e5e4" xblock-family="xblock.v1"
+            discussion_id="{discussion_id}"
+            discussion_category="{category}"
+            discussion_target="{target}"
+        />
+        """.format(discussion_id=discussion_id, category=category, target=target)
+        node = self._make_xml_node(xblock_xml)
+
+        patched_load_definition_xml.side_effect = Exception("Irrelevant")
+
+        block = DiscussionXBlock.parse_xml(node, self.runtime_mock, self.keys, self.id_gen_mock)
+        self.assertEqual(block.discussion_id, discussion_id)
+        self.assertEqual(block.discussion_category, category)
+        self.assertEqual(block.discussion_target, target)
+
+    @mock.patch(DISCUSSION_XBLOCK_LOCATION + ".load_definition_xml")
+    @ddt.unpack
+    @ddt.data(
+        ("ID1", "Test category", "Test target"),
+        ("ID2", "Some other category", "Some other target")
+    )
+    def test_legacy_export_format(self, discussion_id, category, target, patched_load_definition_xml):
+        """
+        Test that legacy export XML format can be parsed preserving field values
+        """
+        xblock_xml = """
+            <discussion url_name="82bb87a2d22240b1adac2dfcc1e7e5e4" discussion_id="{discussion_id}"/>
+        """.format(discussion_id=discussion_id)
+        xblock_definition_xml = """
+            <discussion discussion_category="{category}" discussion_target="{target}"/>
+        """.format(category=category, target=target)
+        node = self._make_xml_node(xblock_xml)
+        definition_node = self._make_xml_node(xblock_definition_xml)
+
+        patched_load_definition_xml.return_value = (definition_node, "irrelevant")
+
+        block = DiscussionXBlock.parse_xml(node, self.runtime_mock, self.keys, self.id_gen_mock)
+        self.assertEqual(block.discussion_category, category)
+        self.assertEqual(block.discussion_target, target)
+
+    def test_export_default_discussion_id(self):
+        """
+        Test that default discussion_id values are not exported
```

Comment:
I think this should also say, "This is because historically, the OLX format does not include discussion ID values; instead, the IDs are generated deterministically based on the course ID and the usage ID. Changing that ID serialization behavior can cause data loss when switching between an older and newer export, in a course with a course ID different from the one from which the export was created - because the discussion ID would be different."